tests/tests.py

```diff
@@ -271,18 +297,23 @@ def parse_args():
     else:
         gdbinit_path = os.path.join(root_dir, "gdbinit.py")
 
-    gdb_binary = "gdb"
+    test_dir_path = TEST_FOLDER_NAME[args.type]
 
     if args.type == "gdb":
-        ensureZigPath()
-        makeBinaries()
-    else:
-        makeCrossArchBinaries()
+        gdb_binary = "gdb"
+        ensure_zig_path()
+        make_binaries(test_dir_path)
+    elif args.type == "cross-arch":
         gdb_binary = "gdb-multiarch"
+        make_binaries(test_dir_path)
+    else:
+        raise NotImplementedError(args.type)
 
-    test_dir_path = TEST_FOLDER_NAME[args.type]
-    tests_list = getTestsList(
+    tests_list = get_tests_list(
         args.collect_only, args.test_name_filter, gdb_binary, gdbinit_path, test_dir_path
     )
-    ports = open_ports(len(tests_list))
```

Comment:
Why was open_ports removed?

Reply:
new function for malloc random port introduced `reserve_port` :)